bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/gtk/org/eclipse/swt/dnd/TextTransfer.java

```diff
@@ -145,13 +147,13 @@ protected int[] getTypeIds() {
 	if(GTK.GTK4) {
 		return new int[] {(int) OS.G_TYPE_STRING()};
 	}
-	return new int[] {UTF8_STRING_ID, STRING_ID};
+	return new int[] {UTF8_STRING_ID, STRING_ID, TEXTPLAINUTF8_ID};
 }
 
 @Override
 protected String[] getTypeNames() {
 	if (OS.isX11()) {
-		return new String[] {UTF8_STRING, COMPOUND_TEXT, STRING};
+		return new String[] {UTF8_STRING, COMPOUND_TEXT, STRING, TEXTPLAINUTF8};
```

Comment:
maybe I misunderstood something, but `OS.isX11()` means we are not using Wayland - the PR title says the opposite?

Reply:
Thanks for the catch you are right.
I have tested the commit first on the R4_29_maintenance branch
With this original commit: https://github.com/eclipse-platform/eclipse.platform.swt/compare/master...the-snowwhite:eclipse.platform.swt:wayland_qt_paste_fix_R4.29
Reason I was then able to try the fix on the current versions of Eclipse (2023.9)  on both gtk v.3.24.33 and v3.24.38
via just replacing the 3 swt plugin binaries. on my 2 Neon workstations.
this way:
https://github.com/eclipse-platform/eclipse.platform.swt/discussions/858
Later I migrated the changes by hand to master as eclipse reported warninge about code changes when i tried to pathh the commit to master(due to me being very bad to resolve merge conflicts in eclipse).
So I and made this paste mistake.
I have updated and force pushed the branch is that ok ?